### ISSUE TITLE
Bracket pair colorization mistakenly colors angle brackets in syntax

### ISSUE THREAD

**Issue by Spanfile** [microsoft/vscode#131176]
Issue Type: <b>Bug</b>

Bracket pair colorization mistakenly pairs two unrelated bitshift operators together. It also colors things like comparisons, return values and "thick" arrows.

![image](https://user-images.githubusercontent.com/5912012/130077325-2edf47d0-e01f-41e4-b579-9d3e8106d45c.png)
![image](https://user-images.githubusercontent.com/5912012/130077652-4226f59f-16ce-4435-8cf7-67498c83cf14.png)
![image](https://user-images.githubusercontent.com/5912012/130077728-d0e46b45-b103-47f9-9a27-99a8ed49fbfc.png)
![image](https://user-images.githubusercontent.com/5912012/130077794-b8d46a0e-5ecc-4c12-baab-c720b1479f89.png)

VS Code version: Code - Insiders 1.60.0-insider (c08913337cb8b95493be8e99d62e0b3241563f6b, 2021-08-19T05:15:56.182Z)
OS version: Linux x64 5.13.12-xanmod1-span
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz (8 x 3600)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|0, 0, 1|
|Memory (System)|31.29GB (3.54GB free)|
|Process Argv|--unity-launch --crash-reporter-id bd9b53b6-bf9d-4112-a5f5-080f8f241743|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (15)</summary>

Extension|Author (truncated)|Version
---|---|---
toml|be5|0.5.1
better-toml|bun|0.3.2
vscode-lua-format|Koi|1.3.8
rust-analyzer|mat|0.4.0-dev
vscode-docker|ms-|1.16.0
python|ms-|2021.8.1105858891
vscode-pylance|ms-|2021.8.1
jupyter|ms-|2021.8.2028630014
LiveServer|rit|5.6.1
crates|ser|0.5.9
vscode-zipexplorer|sle|0.3.1
lua|sum|2.3.7
vscode-lldb|vad|1.6.5
markdown-all-in-one|yzh|3.4.0
vscode-proto3|zxh|0.5.4

</details>

**Comment by hediet** [comment]
That is because the [rust language defined `<` ... `>` as bracket pair](https://github.com/rust-lang/vscode-rust/blob/master/language-configuration.json#L10).

@alexdima do you have ideas how to fix this?

1. Have a global language override for bracket pair colorization
2. Ask the rust extension to remove `<`...`>`
3. Remove `<` ... `>` when doing bracket pair colorization for every language (as this is also [problematic for jsx](#131003))
4. ?

**Comment by IllusionMH** [comment]
Is it possible to add language specific overrides for brackets pairs in config?
```json
"[typescriptreact]": {
    "editor.bracketPairColorization.pairs": [
        ["{", "}"],
        ["[", "]"],
        ["(", ")"]
    ]
}
```

Or other way to specify subset of pairs provided by language. Or expend it if implementation supports expanding list of pairs and don't rely on language implementation.

**Comment by hediet** [comment]
This issue would be fixed by either #131412 or #131194.
